ADALiOS/ADALiOS/ADKeychainItem.m

```diff
@@ -0,0 +1,366 @@
+//
+//  ADKeychainItem.m
+//  ADALiOS
+//
+//  Created by Ryan Pangrle on 8/18/15.
+//  Copyright (c) 2015 MS Open Tech. All rights reserved.
+//
+
+#import "ADKeychainItem.h"
+#import "ADTokenCacheStoreItem+Internal.h"
+#import "ADProfileInfo.h"
+
+@interface ADKeychainToken : NSObject <NSCoding, NSSecureCoding>
+
+@property NSSet* scopes;
+
+@property NSString* accessToken;
+@property NSString* accessTokenType;
+@property NSDate* expiresOn;
+
+- (void)addToTokenItem:(ADTokenCacheStoreItem*)item;
+
+@end
+
+@interface ADKeychainPolicyItem : NSObject <NSCoding, NSSecureCoding>
+
+@property NSString* refreshToken;
+
+- (void)addAccessTokenWithScopes:(NSSet*)scopes
+                          toItem:(ADTokenCacheStoreItem*)item;
+
+@end
+
+@implementation ADKeychainToken
+
+- (id)initWithCoder:(NSCoder*)aDecoder
+{
+    if (!(self = [super init]))
+    {
+        return nil;
+    }
+    
+    _scopes = [aDecoder decodeObjectOfClass:[NSSet class] forKey:@"scopes"];
+    _accessToken = [aDecoder decodeObjectOfClass:[NSString class] forKey:@"accessToken"];
```

Comment:
He was referring to accessToken vs idToken. In convergence world you get an idtoken by asking for just your client id scope.